net/mwan3/files/lib/mwan3/mwan3.sh

```diff
@@ -37,89 +38,163 @@ MM_UNREACHABLE=""
 command -v ip6tables > /dev/null
 NO_IPV6=$?
 
-# return true(=0) if has any mwan3 interface enabled
-# otherwise return false
-mwan3_rtmon_ipv4()
+SCRIPTNAME="$(basename "$0")"
+LOG()
 {
-	local idx=0
-	local ret=1
-	local tbl=""
-
-	local tid family enabled
-
-	mkdir -p /tmp/mwan3rtmon
-	($IP4 route list table main  | grep -v "^default\|linkdown" | sort -n; echo empty fixup) >/tmp/mwan3rtmon/ipv4.main
-	while uci get mwan3.@interface[$idx] >/dev/null 2>&1 ; do
-		tid=$((idx+1))
-
-		family="$(uci -q get mwan3.@interface[$idx].family)"
-		[ -z "$family" ] && family="ipv4"
-
-		enabled="$(uci -q get mwan3.@interface[$idx].enabled)"
-		[ -z "$enabled" ] && enabled="0"
-
-		[ "$family" = "ipv4" ] && {
-			tbl=$($IP4 route list table $tid 2>/dev/null)
-			if echo "$tbl" | grep -q ^default; then
-				(echo "$tbl"  | grep -v "^default\|linkdown" | sort -n; echo empty fixup) >/tmp/mwan3rtmon/ipv4.$tid
-				cat /tmp/mwan3rtmon/ipv4.$tid | grep -v -x -F -f /tmp/mwan3rtmon/ipv4.main | while read line; do
-					$IP4 route del table $tid $line
-				done
-				cat /tmp/mwan3rtmon/ipv4.main | grep -v -x -F -f /tmp/mwan3rtmon/ipv4.$tid | while read line; do
-					$IP4 route add table $tid $line
-				done
-			fi
-		}
-		if [ "$enabled" = "1" ]; then
-			ret=0
+	local facility=$1; shift
+	# in development, we want to show 'debug' level logs
+	# when this release is out of beta, the comment in the line below
+	# should be removed
+	# [ "$facility" = "debug" ] && continue
+	logger -t "$SCRIPTNAME[$$]" -p $facility "$*"
+}
+
+
+mwan3_push_update()
+{
+	# helper function to build an update string to pass on to
+	# IPTR or IPS RESTORE. Modifies the 'update' variable in
+	# the local scope.
+	update="$update
+$*";
+}
+
+mwan3_rtmon_route_handle()
```

Comment:
Should we not move this line to rtmon?

Reply:
Yes, better to put it there. Will fix this evening.

Reply:
remove it(mwan3_rtmon_route_handle) from mwan3.sh? now dup